### PR TITLE
Adding Game Menu

### PR DESCRIPTION
**Pull request by PondWader**
This pull request is for adding a game menu.

The game menu looks as follows:
[here](https://imgur.com/a/obgC9MM)

It is opened by pressing escape, often the player will have to press escape twice as when the user is in mouse lock it doesn't register as a key press, I do not see this as a major problem and I guess it's just a disadvantage of having it in the browser. I did consider doing something with the mouse lock toggle event.

Options is not completed right now, my plan for that is to add the ability to edit controls and save the player's preferences in local storage but didn't want to go ahead with that until I have you guys approval.

Disconnect simply refreshes the page to let the player join a new server or rejoin.

I tried to use the same style as everything else I could find, this is my first ever attempt at contributing to an open source project. So yeah, I'm open to feedback!
GameMenu

### CODE REVIEW

**Comment by AwesomestCode** [comment]
fixed a linking issue in the PR description (it linked to localhost earlier)

**Review by AwesomestCode — DISMISSED** [review]
Might want to put this commit in a separate PR

File: index.js [hunk 1]
```diff
@@ -306,7 +309,9 @@ async function connect (options) {
       renderer.domElement.mozRequestPointerLock ||
       renderer.domElement.webkitRequestPointerLock
     document.addEventListener('mousedown', (e) => {
-      renderer.domElement.requestPointerLock()
```
Comment (AwesomestCode) [inline index.js:309]: Might want to put this in a separate PR

File: lib/gameMenu.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    render() {
+        return html`
+    <github-link></github-link>
+    <div class="menu-box">
+      <h2 class="title">Game Menu</h2>
+        <div class="spacev"></div>
+        <legacy-button btn-width="100%" @click=${() => { this.disableGameMenu() }}>Back to Game</legacy-button>
+        <div class="spacev"></div>
+        <legacy-button btn-width="100%" onClick="alert('This feature is not available yet!');">Options</legacy-button>
```
Comment (AwesomestCode) [inline lib/gameMenu.js:103]: I sort of feel like alert popups aren't very user friendly: it might be better to use a custom modal

File: lib/gameMenu.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        <div class="spacev"></div>
+        <legacy-button btn-width="100%" onClick="window.location.reload();">Disconnect</legacy-button>
+    </div>
+    `
+    }
+
+    init () {
+
```
Comment (AwesomestCode) [inline lib/gameMenu.js:111]: please lint by running ``npm run fix``` and then pushing

**Review by AwesomestCode — CHANGES_REQUESTED** [review]
Looks good, just some minor edits. Thanks for your contribution 😄

File: lib/gameMenu.js (hunk 2 above)
Comment (PondWader) [reply lib/gameMenu.js:103]: I agree, this is in no way meant to be a permanent thing.

File: index.js (hunk 1 above)
Comment (PondWader) [reply index.js:309]: Maybe, however the edit is required for the game menu to function over all, so might as well add it for the chat too?

File: lib/gameMenu.js (hunk 3 above)
Comment (PondWader) [reply lib/gameMenu.js:111]: Will do

File: lib/gameMenu.js (hunk 2 above)
Comment (AwesomestCode) [reply lib/gameMenu.js:103]: Do you mean that this PR isn’t finished, or that you’ll create another PR later with the options menu?

Comment (PondWader) [reply lib/gameMenu.js:103]: I'll probably make another PR later to complete options, however I can make the alert better before this is merged.

File: index.js (hunk 1 above)
Comment (PondWader) [reply index.js:309]: What do you think? Keep it here or put it in a separate PR, I will need to do it for when the user is in the game menu anyway, so felt natural to include chat with it in this PR. But if you believe the chat bit should be in a separate PR I will do that.

Comment (rom1504) [reply index.js:309]: Let's keep it here

**Comment by PondWader** [comment]
> Looks good, just some minor edits are needed. Thanks for your contribution 😄

I believe I have completed the requested edits. 👍

**Comment by rom1504** [comment]
Lint is failing you need to run npm run fix

**Comment by PondWader** [comment]
> Lint is failing you need to run npm run fix

Should all be fixed now.

File: lib/gameMenu.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  render () {
+    return html`
+    <github-link></github-link>
+    <div class="menu-box">
+      <h2 class="title">Game Menu</h2>
+        <div class="spacev"></div>
+        <legacy-button btn-width="100%" @click=${() => { this.disableGameMenu() }}>Back to Game</legacy-button>
+        <div class="spacev"></div>
+        <legacy-button btn-width="100%">Options</legacy-button>
```
Comment (AwesomestCode) [inline lib/gameMenu.js:104]: Personally I'd still prefer showing a modal (e.g. overlaying an image on top of the screen saying that it isn't implemented yet) so people don't think it's broken; though don't use alerts. but if Rom's fine I am too.

**Review by AwesomestCode — APPROVED** [review]
Haven't tried it out but the code LGTM

File: lib/gameMenu.js (hunk 4 above)
Comment (PondWader) [reply lib/gameMenu.js:104]: I will add it later today, I wasn't sure whether to go ahead with it or not.

**Review by rom1504 — APPROVED** [review]
